File: src/test/java/refactown/cleancode/c08boundary/java/LogonControllerTest.java
package refactown.cleancode.c08boundary.java;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class LogonControllerTest {

    @Test
    public void logonManualDependency() {
        User userMock = new User("admin", "admin");
        GsonJsonParser parser = new GsonJsonParser();
        LogonController controller = new LogonController(parser);
        User user = controller.logon("admin", "123");

        assertEquals(userMock.getLogin(), user.getLogin());
        assertEquals(userMock.getRole(), user.getRole());
    }

    @Test
    public void  logonInjectDependency() {
        User userMock = new User("admin", "admin");
        LogonController controller = inject(LogonController.class);
        User user = controller.logon("admin", "123");

        assertEquals(userMock.getLogin(), user.getLogin());
        assertEquals(userMock.getRole(), user.getRole());

    }

	private LogonController inject(Class<LogonController> class1) {
		return new LogonController(new GsonJsonParser());
	}

}

File: src/test/kotlin/refactown/cleancode/c08boundary/LogonControllerTest.kt
package refactown.cleancode.c08boundary

import org.junit.jupiter.api.*

import org.junit.jupiter.api.Assertions.*

internal class LogonControllerTest {


    @Test
    fun logonManualDependency() {
        val userMock = User("admin", "admin")
        val parser = GsonJsonParser()
        val controller = LogonController(parser)
        val user = controller.logon("admin", "123")

        assertEquals(userMock, user)
    }

    @Test
    fun logonInjectDependency() {
        val userMock = User("admin", "admin")
        val controller = inject(LogonController::class.java) as LogonController
        val user = controller.logon("admin", "123")

        assertEquals(userMock, user)
    }

}

    fun inject(clazz: Class<*>): Any{
        return LogonController(GsonJsonParser())
    }
